### PR TITLE
feat(cdk): focusable list & item upgrade, focusable grid feature and docs

### PR DESCRIPTION
**Pull request by platon-rov**
## Related Issue(s)

Closes https://github.com/SAP/fundamental-ngx/issues/9202

## Description

* Focusable Grid feature for CDK and docs for it.
* Focusable Item enhancements, refactoring
  * Functionality to focus inner tabbable elements.
  * Announcing info when in grid.
* Focusable List enhancements, refactoring
  * Keeping only one tab stop per list.
  * Input to manipulate wrapping mode.

### After:

image

### CODE REVIEW

File: libs/docs/cdk/focusable-grid/focusable-grid-header/focusable-grid-header.component.html [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<header>Focusable Grid</header>
+<description>
+    You can quickly build out focusable grid collection for your components. For that you will need to use a combination
+    of <code>[fdkFocusableGrid]</code>, <code>[fdkFocusableList]</code> and <code>[fdkFocusableItem]</code> directives.
+    It also respects <code>[fdkDisabled]</code> and <code>[fdkReadonly]</code> directives, so that those elements will
```
Comment (mikerodonnell89) [inline libs/docs/cdk/focusable-grid/focusable-grid-header/focusable-grid-header.component.html:5]: Typical behavior for readonly elements (such as inputs) still allows them to be focused and reached via tab keys - so maybe they should not be excluded from the arrow keys as well?

File: libs/docs/cdk/focusable-grid/examples/short-rows-example/short-rows-example.component.html [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<div fdkFocusableGrid shortRowFocus="first">
+    <div *ngFor="let list of [].constructor(5); let rowIndex = index" fdkFocusableList navigationDirection="horizontal">
```
Comment (mikerodonnell89) [inline libs/docs/cdk/focusable-grid/examples/short-rows-example/short-rows-example.component.html:2]: Confused at the purpose of `navigationDirection="horizontal"` here, isn't the direction both horizontal and vertical?

File: libs/cdk/src/lib/utils/directives/focusable-list/focusable-list.directive.ts [hunk 3]
```diff
@@ -26,22 +60,54 @@ export class DeprecatedFocusableListDirective extends DeprecatedSelector {}
...
+        {
+            provide: FDK_FOCUSABLE_LIST_DIRECTIVE,
+            useExisting: FocusableListDirective
+        },
+        DestroyedService
+    ]
 })
 export class FocusableListDirective implements AfterViewInit {
-    /** Direction of navigation. */
+    /** Direction of navigation. Always horizontal when in grid. */
```
Comment (mikerodonnell89) [inline libs/cdk/src/lib/utils/directives/focusable-list/focusable-list.directive.ts:72]: In my opinion a third option here like `'grid'` would make more sense

File: libs/cdk/src/lib/utils/directives/focusable-item/focusable-item.directive.ts [hunk 4]
```diff
@@ -38,64 +43,152 @@ export class DeprecatedFocusableItemDirective extends DeprecatedSelector {}
...
+            this._disableTabbableElements();
+        }
     }
 
     /** @hidden */
-    ngAfterViewInit(): void {
-        this._viewInit$.next(true);
+    @HostListener('focusin')
+    async _onFocusin(): Promise<void> {
+        if (this._timerId != null) {
```
Comment (mikerodonnell89) [inline libs/cdk/src/lib/utils/directives/focusable-item/focusable-item.directive.ts:138]: What is the purpose of `_timerId`? I'm sure there's a reason I just don't know. Change detection related?

**Review by mikerodonnell89 — COMMENTED** [review]
This is great! Couple comments/questions

File: libs/cdk/src/lib/utils/directives/focusable-list/focusable-list.directive.ts (hunk 3 above)
Comment (platon-rov) [reply libs/cdk/src/lib/utils/directives/focusable-list/focusable-list.directive.ts:72]: Good idea!

File: libs/cdk/src/lib/utils/directives/focusable-item/focusable-item.directive.ts (hunk 4 above)
Comment (platon-rov) [reply libs/cdk/src/lib/utils/directives/focusable-item/focusable-item.directive.ts:138]: That's for supporting multiple focusable elements inside a cell, when we come to the cell we should focus first focusable child and read a11y text but then when tabbing between focusable elements inside a single cell we should not manipulate focus neither read a11y text again.

File: libs/docs/cdk/focusable-grid/focusable-grid-header/focusable-grid-header.component.html (hunk 1 above)
Comment (platon-rov) [reply libs/docs/cdk/focusable-grid/focusable-grid-header/focusable-grid-header.component.html:5]: Yea, I agree with you, let me change it here and in focusable list.

**Review by mikerodonnell89 — CHANGES_REQUESTED** [review]
build is failing: https://app.netlify.com/sites/fundamental-ngx/deploys/63ca8cd1b7845b00091e8970#L1063

File: libs/cdk/src/lib/utils/directives/focusable-list/focusable-list.directive.ts [hunk 5]
```diff
@@ -117,12 +117,12 @@ export class FocusableListDirective implements OnChanges, AfterViewInit {
             return;
         }
 
-        if (changes.wrap) {
+        if (changes['wrap']) {
             this._keyManager = this._keyManager.withWrap(changes.wrap.currentValue);
```
Comment (g-cheishvili) [inline libs/cdk/src/lib/utils/directives/focusable-list/focusable-list.directive.ts:121]:
```ts
this._keyManager = this._keyManager.withWrap(changes['wrap'].currentValue);
```

**Review by g-cheishvili — APPROVED** [review]
Looks good to me

**Review by mikerodonnell89 — CHANGES_REQUESTED** [review]
This looks and works great but just one last thing, after mouse clicking a focusable item, the item is focused but the focus outline doesn't initially appear, I can only see it once I start using the keyboard

**Comment by platon-rov** [comment]
> This looks and works great but just one last thing, after mouse clicking a focusable item, the item is focused but the focus outline doesn't initially appear, I can only see it once I start using the keyboard

Nice catch, so I've updated docs and now it looks better, although this feature is not about styling so when using it - styles for focus state should be implemented on consumer side.

**Pull request merged**